include/SFML/Window/Event.hpp

```diff
@@ -34,196 +34,298 @@
-        MouseEntered,           //!< The mouse cursor entered the area of the window (no data)
-        MouseLeft,              //!< The mouse cursor left the area of the window (no data)
-        JoystickButtonPressed,  //!< A joystick button was pressed (data in event.joystickButton)
-        JoystickButtonReleased, //!< A joystick button was released (data in event.joystickButton)
-        JoystickMoved,          //!< The joystick moved along an axis (data in event.joystickMove)
-        JoystickConnected,      //!< A joystick was connected (data in event.joystickConnect)
-        JoystickDisconnected,   //!< A joystick was disconnected (data in event.joystickConnect)
-        TouchBegan,             //!< A touch event began (data in event.touch)
-        TouchMoved,             //!< A touch moved (data in event.touch)
-        TouchEnded,             //!< A touch event ended (data in event.touch)
-        SensorChanged,          //!< A sensor value changed (data in event.sensor)
-
-        Count //!< Keep last -- the total number of event types
+        Empty,                  //!< Default type
+        Closed,                 //!< The window requested to be closed
+        Resized,                //!< The window was resized
+        LostFocus,              //!< The window lost the focus
+        GainedFocus,            //!< The window gained the focus
+        TextEntered,            //!< A character was entered
+        KeyPressed,             //!< A key was pressed
+        KeyReleased,            //!< A key was released
+        MouseWheelScrolled,     //!< The mouse wheel was scrolled
+        MouseButtonPressed,     //!< A mouse button was pressed
+        MouseButtonReleased,    //!< A mouse button was released
+        MouseMoved,             //!< The mouse cursor moved
+        MouseEntered,           //!< The mouse cursor entered the area of the window
+        MouseLeft,              //!< The mouse cursor left the area of the window
+        JoystickButtonPressed,  //!< A joystick button was pressed
+        JoystickButtonReleased, //!< A joystick button was released
+        JoystickMoved,          //!< The joystick moved along an axis
+        JoystickConnected,      //!< A joystick was connected
+        JoystickDisconnected,   //!< A joystick was disconnected
+        TouchBegan,             //!< A touch event began
+        TouchMoved,             //!< A touch moved
+        TouchEnded,             //!< A touch event ended
+        SensorChanged,          //!< A sensor value changed
     };
 
+    Event() = default;
+
+    template <typename T>
+    explicit Event(const T& t) : m_data(t)
+    {
+    }
+
+    [[nodiscard]] Type getType() const
+    {
+        return static_cast<Type>(m_data.index());
+    }
+
+    template <typename T>
+    [[nodiscard]] bool is() const
+    {
+        return std::holds_alternative<T>(m_data);
+    }
+
+    template <typename T>
+    void set(const T& t)
```

Comment:
Is the idea to then depend on the compiler-generated assignment operators?

Reply:
Agree, this might be unnecessary, but as long as there's a `get()` (name wise), having a `set()` (even with duplicated functionality) doesn't sound inherently wrong to me.